templates/server-discovery-role.yaml

```diff
@@ -1,7 +1,8 @@
 {{ template "vault.mode" . }}
-{{- if ne .mode "external" }}
-{{- if .serverEnabled -}}
+{{ template "vault.serverServiceAccountEnabled" . }}
 {{- if eq .mode "ha" }}
+{{- if .serverEnabled -}}
+{{- if .serverServiceAccountEnabled -}}
```

Comment:
I think disabling the discovery role and binding if the server isn't enabled makes sense, but I think we may want to still create them even if the chart isn't creating the service account. Just thinking that the service account that the user supplies via `server.serviceAccount.name` will still need these capabilities and it's convenient if the chart just adds them for the user.

Reply:
Yeah that's fair, I'll add a net-new config option to disable it instead